docs/odh-rhoai/stable_diffusion_safety_checker.md

```diff
@@ -0,0 +1,43 @@
+# Stable Diffusion Safety Checker
+
+## Image Guardrails
+
+When generating images, you want to be able to check whether your model's outputs are safe to display and block them if they are not.  
+This is where image generation guardrails come into play. By using image models trained on unsafe content, we can classify the output images to ensure they comply with our guidelines.  
+Without these kinds of safeguards, users would be able to generate unethical and harmful content from our models.  
+
+![safety-checker-flow.png](./img/safety-checker-flow.png)
+
+### **Key risks when not using guardrails:**  
+- Misinformation and Deepfakes – The ability to create realistic but fake images, which may be used to spread misinformation or deceive people.
+- Harmful and Inappropriate Content – The generation of violent, explicit, or illegal material.
+- Intellectual Property and Copyright Infringement – The replication of or close resemblance to copyrighted content, leading to potential legal and ethical issues.
+
+## Stable Diffusion Safety Checker
+
+The Stable Diffusion Safety Checker is one of these image guardrails, specifically built for analyzing the outputs of diffusion models.  
+It allows application developers to check any images generated by a Stable Diffusion model before displaying them to end-users.
+
+The model it uses is the [Stable Diffusion Safety Checker](https://huggingface.co/CompVis/stable-diffusion-safety-checker) which is based on a fine-tuned [CLIP model](https://huggingface.co/openai/clip-vit-large-patch14) under the [MIT License](https://github.com/openai/CLIP/blob/main/LICENSE).  
+
+### How does it work?   
+
+The Stable Diffusion Safety Checker compares embeddings (a list of values that represent the image) generated by the model with predefined embeddings that represent harmful images. If the output embedding is too close to any of the predefined embeddings, the model will output `True`, meaning that the image may be harmful. Otherwise, it will output `False`.  
+You can find the full code in this [HuggingFace repo](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/safety_checker.py).
+
+### How can you customize it?
+
+Since the model simply evaluates how close an image is to different embeddings, a straightforward way to customize it is to add a new predefined embedding of an image you ran through the model. This will cause the model to return `True` for all images that are similar to the new embedding.  
+However, note that while this method works, it has limitations in quality. For improved accuracy when adding new categories, fine-tuning the model is recommended.
+
+## Example of using it
+
```

Comment:
Can you move/copy this code to https://github.com/rh-aiservices-bu/image-generation-on-openshift  so that we have everything at the same place?
Also, in this code, you are pointing to some Routes on the Dev cluster. You should change them to env vars for the URL and the API key that will go with it. Like this people can directly use the code against MaaS.

Reply:
It's now been moved there :)